HpToolsLauncher/ParallelRunner/ParallelRunnerEnvironmentUtil.cs

```diff
@@ -183,316 +182,325 @@ public static MobileEnvironment ParseMobileEnvironment(string environment)
+
+                var item = new ParallelTestRunConfigurationItem
+                {
+                    test = testInfo.TestPath,
+                    env = environment,
+                    reportPath = testInfo.TestPath
+                };
+
+                items.Add(item);
+            }
+
+            parallelTestRunConfiguration.parallelRuns = items.ToArray();
+
+            return parallelTestRunConfiguration;
+        }
+
+        /// <summary>
+        /// Return the proxy settings.
+        /// </summary>
+        /// <param name="mcConnectionInfo">the mc connection info</param>
+        /// <returns></returns>
+        public static ProxySettings GetMCProxySettings(McConnectionInfo mcConnectionInfo)
+        {
+            if (String.IsNullOrEmpty(mcConnectionInfo.MobileProxySetting_Address))
+                return null;
+
+            AuthenticationSettings authenticationSettings = null;
+
+            if (!string.IsNullOrEmpty(mcConnectionInfo.MobileProxySetting_UserName)
+                && !string.IsNullOrEmpty(mcConnectionInfo.MobileProxySetting_Password))
+            {
+                authenticationSettings = new AuthenticationSettings
+                {
+                    username = mcConnectionInfo.MobileProxySetting_UserName,
+                    password = WinUserNativeMethods.
+                        ProtectBSTRToBase64(mcConnectionInfo.MobileProxySetting_Password)
+                };
+            }
+
+            ProxySettings proxySettings = new ProxySettings
+            {
+                authentication = authenticationSettings,
+                hostname = mcConnectionInfo.MobileProxySetting_Address,
+                port = mcConnectionInfo.MobileProxySetting_Port,
+                type = mcConnectionInfo.MobileProxyType == 1 ? "system" : "http",
+            };
+
+            return proxySettings;
+        }
+
+        /// <summary>
+        /// Parses the MC settings and returns the corresponding UFT settings.
+        /// </summary>
+        /// <param name="info"> the mc settings</param>
+        /// <returns> the parallel runner uft settings </returns>
+        public static UFTSettings ParseMCSettings(McConnectionInfo info)
+        {
+            if (string.IsNullOrEmpty(info.MobileHostAddress) ||
+                (string.IsNullOrEmpty(info.MobileUserName) && string.IsNullOrEmpty(info.MobileClientId)) ||
+                (string.IsNullOrEmpty(info.MobilePassword) && string.IsNullOrEmpty(info.MobileSecretKey)) ||
+                string.IsNullOrEmpty(info.MobileHostPort))
+                return null;
+
+            MCSettings mcSettings = new MCSettings
+            {
+                authType = (int)info.MobileAuthType,
+                hostname = info.MobileHostAddress,
+                port = Convert.ToInt32(info.MobileHostPort),
+                protocol = info.MobileUseSSL > 0 ? "https" : "http"
+            };
+            if (info.MobileAuthType == (int)McConnectionInfo.AuthType.UsernamePassword)

```

Comment:
You need to give the `AuthType` enum values a fixed int value, for each enum, otherwise, here you convert to int, it may be problematic if the order of the enum values are changed
```C#
public enum AuthType
        {
            UsernamePassword,
            AuthToken
        }
```